pkg/client/log.go

```diff
@@ -31,7 +34,7 @@ func GetLogger(format string, debug bool) (*zap.Logger, error) {
 }
 
 func FlushBufferOnExit(logger *zap.Logger) {
-	if err := logger.Sync(); err != nil {
+	if err := logger.Sync(); err != nil && !errors.Is(err, syscall.ENOTTY) {
```

Comment:
what does this `ENOTTY` mean?

Reply:
we need this for when stdout/stderr is redirected to the console, such as during unit tests

this is a known issue with Zap unfortunately: https://github.com/uber-go/zap/issues/880#issuecomment-1181854418

Reply:
add a note for reference?

Reply:
will add a clarifying comment

Reply:
and maybe refine the code a bit to exit early

```
syncErr := logger.Sync()
if syncErr == nil {
  return
}

switch {
  case errors.Is(err, syscall.ENOTTY):
    // known issue
    return
  default:
    // log
}
```